From: Matthew Garrett <mjg@redhat.com>
Date: Thu, 20 Sep 2012 10:41:03 -0400
Subject: [PATCH V2 08/10] efi: Enable secure boot lockdown automatically when enabled in firmware
Patch-mainline: Not yet, reviewing
References: fate#314486
Target: SLE-11 SP3

The firmware has a set of flags that indicate whether secure boot is enabled
and enforcing. Use them to indicate whether the kernel should lock itself
down.

Signed-off-by: Matthew Garrett <mjg@redhat.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
Automatically created from "patches.suse/0008_efi:_Enable_secure_boot_lockdown_automatically_when_enabled_in_firmware_v2.patch" by xen-port-patches.py

--- sle11sp3.orig/arch/x86/kernel/setup-xen.c	2013-02-05 16:39:12.000000000 +0100
+++ sle11sp3/arch/x86/kernel/setup-xen.c	2013-02-05 16:50:11.000000000 +0100
@@ -1184,6 +1184,13 @@ void __init setup_arch(char **cmdline_p)
 
 	io_delay_init();
 
+#ifndef CONFIG_XEN
+	if (boot_params.secure_boot)
+#else
+	if (0)
+#endif
+		secureboot_enable();
+
 #ifdef CONFIG_ACPI
 	if (!is_initial_xendomain()) {
 		printk(KERN_INFO "ACPI in unprivileged domain disabled\n");
